src/servers/src/grpc/region_server.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use api::helper::region_request_type;
+use api::v1::auth_header::AuthScheme;
+use api::v1::region::region_request::Request as RequestBody;
+use api::v1::region::region_server_server::RegionServer as RegionServerService;
+use api::v1::region::{RegionRequest, RegionResponse};
+use api::v1::{Basic, RequestHeader};
+use async_trait::async_trait;
+use auth::{Identity, Password, UserInfoRef, UserProviderRef};
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_catalog::parse_catalog_and_schema_from_db_string;
+use common_error::ext::ErrorExt;
+use common_runtime::Runtime;
+use common_telemetry::{debug, error};
+use metrics::increment_counter;
+use session::context::{QueryContextBuilder, QueryContextRef};
+use snafu::{OptionExt, ResultExt};
+use tonic::{Request, Response};
+
+use crate::error::{
+    AuthSnafu, InvalidQuerySnafu, JoinTaskSnafu, NotFoundAuthHeaderSnafu, Result,
+    UnsupportedAuthSchemeSnafu,
+};
+use crate::grpc::greptime_handler::RequestTimer;
+use crate::grpc::TonicResult;
+use crate::metrics::{METRIC_AUTH_FAILURE, METRIC_CODE_LABEL};
+
+#[async_trait]
+pub trait RegionServerHandler: Send + Sync {
+    async fn handle(&self, request: RequestBody) -> Result<RegionResponse>;
+}
+
+pub type RegionServerHandlerRef = Arc<dyn RegionServerHandler>;
+
+#[derive(Clone)]
+pub struct RegionServerRequestHandler {
```

Comment:
question(non-blocking): We already define `RegionServerHandler`. Could we rename RegionServerRequestHandler to `RegionRequestHandler`? It handles RegionRequest.

Or rename `RegionServerHandler` to `RequestBodyHandler`...

We can also rename it to `RegionServerServiceImpl`...

Reply:
Going to rename in the follow-up pr